lib/azure/armrest/armrest_collection.rb

```diff
@@ -4,6 +4,7 @@ module Azure
   module Armrest
     class ArmrestCollection < Array
       attr_accessor :continuation_token
+      attr_accessor :response_headers
```

Comment:
I generally avoid redefining initialize on a subclass of a core class because it can lead to unexpected behavior. Since this is only used internally, I don't see the advantage of redefining initialize.